src/main/host/descriptor/regular_file.c

```diff
@@ -87,19 +106,28 @@ static inline RegularFile* _regularfile_legacyFileToRegularFile(LegacyFile* desc
 
 static inline int _regularfile_getOSBackedFD(RegularFile* file) {
     MAGIC_ASSERT(file);
-    return file->osfile.fd;
+    if (file->type != FILE_TYPE_IN_MEMORY)
+        return file->osfile.fd;
+    else
+        // TODO returning 0 would be ambiguous, but also possibly better as caller usually check for nullity, not for OSFILE_INVALID
+        return OSFILE_INVALID;
```

Comment:
Ah right, I was thinking 0 isn't a valid fd, but of course it is.

Ok, +1 for keeping the -1 return value here and filing an issue to clean up if it's impractical to check all the callers in this PR

Reply:
There are 29 easy to fix cases (guarding every file syscall with EBADF). I don't see other problematic cases, I can fix what I see, open an issue and a 2nd pair of eyes can check if I missed anything later.

Reply:
sounds good - looks like you already filed https://github.com/shadow/shadow/issues/2604. I added it to the next release Project so that we hopefully don't forget to fix it before the next release. I'm fine with merging this PR without those fixes in the meantime.

If you'd like to take that on as a follow-up, you're welcome to assign yourself (if GH lets you...?)